.github/workflows/integration-test.yml

```diff
@@ -193,3 +199,48 @@ jobs:
       - name: Run tests through docker image
         run: |
           docker run pycytominer:testing pytest
+      # Gather and prepare the dynamic version and other vars for use with work below.
+      # For example, 1.1.0.post6.dev0+bc093ef becomes 1.1.0.post6.dev0_bc093ef
+      - name: Gather dynamic version and set env var
+        id: env-vars
+        run: >-
+          echo "CLEAN_VERSION=$(echo ${{ env.sdist_filename }} | tr '+' '_' )" >> "$GITHUB_OUTPUT" &&
+          echo "DATE_TAG=$(date +'%y%m%d')" >> "$GITHUB_OUTPUT"
+      - name: Login to Docker Hub
+        # only run this task if we have a pull request merged or the schedule triggers from main
+        if: github.event.pull_request.merged || ( github.event_name == 'schedule' && github.ref == 'refs/heads/main')
+        uses: docker/login-action@v3
+        with:
+          username: ${{ secrets.DOCKER_USER }}
+          password: ${{ secrets.DOCKER_TOKEN }}
+      # Docker image push based on weekly schedule
+      - name: Schedule-based docker image push
+        # only run this task if we have a schedule trigger from main
+        if: github.event_name == 'schedule' &&  github.ref == 'refs/heads/main'
```

Comment:
Is there a case when this would not be `refs/heads/main`? I don't necessarily think so

Reply:
As currently written, we don't have any scheduled action runs that don't target refs/heads/main. However, I think keeping the explicit check for `refs/heads/main` check is good defensive posture to guard against a future scenario where someone working in a branch in a new github actions automation that inadvertently triggers a build and push out to dockerhub.

Reply:
Thanks for the comments here! This was declarative for the reasons Ken mentioned; to avoid any ambiguity on where / when this should run. Thinking about this more, I'm going to add an additional constraint to avoid potential scheduled runs from a forked `main`.

Reply:
I've added in greater specificity here to target only the `cytomining/pycytominer` repository and only `main` for scheduled pushes. Recognize this might be overdoing it but want to avoid any circumstances where we have challenges when it comes to multiple scheduled pushes coming from various locations (potentially overwriting one another without much observability).